### DISCUSSION TITLE
How to get variant SKU using GraphQL getSubscriptions?

### DISCUSSION THREAD

**Discussion by rstml** [swellstores/community#377] [Help] [answered]
I'm using the following GraphQL query to fetch subscriptions. However, `variant.sku` always returns `null` while `id` and `name` are correctly populated.

Similar query for `products` returns variant SKU correctly.

How can I get variant SKU from subscriptions?

```graphql
query getSubscriptions($currency: String) {
  subscriptions(_currency: $currency) {
    results {
      id
      status
      datePeriodEnd
      dateOrderPeriodEnd
      variant {
        name
        sku
        id
      }
      product {
        name
        images {
          id
          caption
          file {
            url
            width
            height
          }
        }
      }
      billingSchedule {
        interval
        intervalCount
        trialDays
      }
      orderSchedule {
        interval
        intervalCount
      }
      recurringTotal
    }
  }
}
```

**Comment by logeshswell** [comment, marked as answer]
You need to open up frontend permissions for your public key as follows:
<img width="514" alt="Screenshot 2023-09-19 at 2 56 03 PM" src="https://github.com/swellstores/community/assets/100766131/456cfa1f-d893-422e-b963-3c89572da19b">

For more details, please refer https://developers.swell.is/guides/opening-content-permissions#adjusting-the-permissions

**Reply by rstml** [reply]
Thanks @logeshswell! "public key" is the key word. Was trying all sorts of permissions on the private key which obviously didn't work. All works now.

Having said that, I found it confusing that by default SKU was available for `products`, but withheld for `subscriptions`.

**Reply by logeshswell** [reply]
Yes, frontend permissions should be set for public keys and NOT for secret keys.
Secret key will have all permissions to make any CRUD operations to any models.

> I found it confusing ...

Yeah, we have some plans to make the UX of the content permissions better in the admin dashboard to avoid such confusion. I will keep you posted if I get some updates on it.
